### PR TITLE
WIP: PLENTY-153 Add basic test functionality

### PR DESCRIPTION
**Pull request by TobiasGraml11**

### CODE REVIEW

**Comment by TobiasGraml11** [comment]
Current status: Travis build successfully run for the first time :tada:

File: spec/buildVueComponents.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    $template = preg_replace('/(<script){1}(.)*(>){1}/', '<template>', $template);
+    $template = str_replace('</script>', '</template>', $template);
+
+    $script = preg_replace('/(Vue.component){1}(\(")(.)*(,){1}/', 'export default', $script);
+    $script = preg_replace('/(\);)$/', '', $script);
+    $script = str_replace('../../../mixins/url', '../../resources/js/src/app/mixins/url', $script);
+    $script = str_replace('../../mixins/url', '../../resources/js/src/app/mixins/url', $script);
+
+    file_put_contents($vue_dir . $component['name'],  $template . '<script>' . $script . '</script>');
```
Comment (dblaichinger) [inline spec/buildVueComponents.php]: In future we should try to use a different way than regex replaces

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    ],
+    transform: {
+        "^.+\\.jsx?$": "babel-jest",
+        '^.+\\.vue$': 'vue-jest',
+    },
+    transformIgnorePatterns: [
+        '/node_modules/'
+    ],
+    globals: {
+        App: {
```
[the remark this answers is not in the document]
Comment (TobiasGraml11) [reply jest.config.js]: I don't think we can get app from Vue.

Vue specific stuff is used with `this` or `Vue.prototype`.

Can't find anything else

File: resources/css/findologic.min.css [hunk 3]
```diff
@@ -1,2 +1,2 @@
...
...r .fl-item-color-tiles-list .fl-item-color-tiles-list-item .fl-color-tile-label .fl-color-tile-background:hover{border:1px solid #ddd}.fl-item-color-tiles-container .fl-item-color-tiles-list .fl-item-color-tiles-list-item .fl-color-tile-label .fl-color-tile-image{position:absolute;left:0;z-index:1;width:30px;height:30px}.fl-item-color-tiles-container .fl-item-color-tiles-list .fl-item-color-tiles-list-item .fl-color-tile-label .fl-color-tile-selected-image{position:absolute;z-index:2}.fl-range-slider-container{display:inline-block}.fl-range-slider-container .fl-range-slider{margin-top:20px}.fl-range-slider-container .fl-range-input{width:50%;height:35px;padding:5px;display:inline}.fl-range-slider-container .fl-range-slider-row{margin-left:-7px;margin-right:-23px}.fl-range-slider-container .fl-unit{margin-left:5px}.fl-range-slider-container .fl-range-slider-submit-btn{margin-top:10px}@media screen and (max-width:543px){.fl-range-slider-container .fl-range-slider-submit-btn{width:100%}.fl-range-slider-container .fl-range-slider-submit-btn>.fa{float:none}}.search-box-inner .autocomplete-suggestions{display:none!important}#fl-promotion{text-align:center}#fl-promotion .image{width:100%}
```
[the remark this answers is not in the document]
Comment (TobiasGraml11) [reply resources/css/findologic.min.css]: As already talked about. Seems like the styles were not built after some changes before this pull request

**Pull request closed without merging**